packages/kit/src/exports/node/index.js

```diff
@@ -94,7 +94,8 @@ function get_raw_body(req, body_size_limit) {
 
 /** @type {import('@sveltejs/kit/node').getRequest} */
 export async function getRequest({ request, base, bodySizeLimit }) {
-	return new Request(base + request.url, {
+	// @ts-expect-error - the request may be a connect request
+	return new Request(base + (request.originalUrl || request.url), {
```

Comment:
Does `request.originalUrl || request.url` serve any purpose other than for Vite's dev mode? I'd prefer not to have any dev-specific code end up in production code built with (say) the Node adapter, which I think this would.

If, on the other hand, this is also helpful for supporting Node adapter output in custom server mode using Connect, that should be made more explicit somewhere. Does it belong in the Node adapter's readme maybe?

Reply:
It helps both dev and preview.

It seems pretty unlikely to me that the values of `url` and `originalUrl` would differ in `adapter-node`. They could if the user mounted the app under a base path themselves in a custom server implementation, but I don't think I'd expect that to work regardless of which value were used here. We could throw an exception if `originalUrl` is defined and does not equal `url` if we want to enforce that they're the same. That'd then give us the ability to explicitly support them being different in the future without it being a breaking change as I really don't know what that'd mean at this point.